synapse/handlers/stats.py

```diff
@@ -52,4 +59,314 @@ def __init__(self, hs):
     def notify_new_event(self):
         """Called when there may be more deltas to process
         """
-        pass
+        if not self.hs.config.stats_enabled:
+            return
+
+        lock = self.store.stats_delta_processing_lock
+
+        @defer.inlineCallbacks
+        def process():
+            yield lock.acquire()
+            try:
+                yield self._unsafe_process()
+            finally:
+                yield lock.release()
+
+        if not lock.locked:
+            # we only want to run this process one-at-a-time,
+            # and also, if the initial background updater wants us to keep out,
+            # we should respect that.
+            try:
+                run_as_background_process("stats.notify_new_event", process)
+            except:  # noqa: E722 – re-raised so fine
+                lock.release()
+                raise
+
+    @defer.inlineCallbacks
+    def _unsafe_process(self):
+        # If self.pos is None then means we haven't fetched it from DB
+        if self.pos is None or None in self.pos.values():
```

Comment:
What does `None in self.pos.values()` mean?

Reply:
Adding a comment.